python/cudf/dataframe/column.py

```diff
@@ -62,6 +62,14 @@ def _concat(cls, objs, dtype=None):
                 dtype = np.dtype(dtype)
                 return Column(Buffer.null(dtype))
 
+        # Find the dtype of each column ignoring pure NULL columns
+        col_dtypes = [o.dtype for o in filter(lambda o: len(o) != o.null_count, objs)]
+        # Use NumPy to find a common dtype
+        common_dtype = np.find_common_type(col_dtypes, [])
```

Comment:
This won't handle category dtypes, so you'd need to check for the category handling before doing this.

Reply:
Ahh I see, is there coercion rules between categories? 
Or should we raise an error when concatenating columns of different category types?

Reply:
I believe we already have logic to synchronize the categories before concatenation, the main thing is we should guard against trying to concatenate a category column or a string column with numeric dtype columns.

Reply:
And in this comment I specifically meant numpy won't know what to do with `"category"` or a `CategoricalDtype` object

Reply:
I see, so it should only try to do casting if all columns are of regular NumPy dtypes?